src/components/AutocompleteSearch/index.jsx

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable react/jsx-props-no-spreading */
+
+import { useCombobox } from 'downshift';
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { fetchAutoCompleteResults } from '../../utils';
+import { loadStatuses } from '../../utils/constants';
+
+function AutocompleteSearch(props) {
+  const { searchTerm, setSearchTerm } = props;
+  const navigate = useNavigate();
+  const [status, setStatus] = useState();
+  const [products, setProducts] = useState([]);
+  const [searchSuggestions, setSearchSuggestions] = useState([]);
+
+  const itemToString = (item) => (item ? item.name : '');
+
+  const {
+    isOpen,
+    getMenuProps,
+    getItemProps,
+    getInputProps,
+    getComboboxProps,
+  } = useCombobox({
+    items: [...searchSuggestions, ...products],
+    itemToString,
+    onInputValueChange: async ({ inputValue }) => {
+      setSearchTerm(inputValue);
+    },
+    onSelectedItemChange({ selectedItem }) {
+      setSearchTerm(selectedItem.value);
+      navigate(`?q=${selectedItem.value}`);
+    },
+  });
+
+  let timeout;
+  useEffect(() => {
+    if (!searchTerm) {
+      setProducts([]);
+      setSearchSuggestions([]);
+      setStatus();
+    } else {
+      (async () => {
+        setStatus(loadStatuses.LOADING);
+        timeout = setTimeout(async () => {
+          try {
+            const res = await fetchAutoCompleteResults(searchTerm);
+            setStatus(loadStatuses.SUCCESS);
+            setProducts(res.sections.Products);
+            setSearchSuggestions(res.sections['Search Suggestions']);
+          } catch (error) {
+            setStatus(loadStatuses.FAILED);
+          }
+        }, 300);
+      })();
+    }
+
+    return () => {
+      window.clearTimeout(timeout);
+    };
+  }, [searchTerm]);
+
+  return (
+    <div className="relative">
```

Comment:
Can we add the data attributes to all these elements? 
https://constructor.slab.com/posts/data-driven-beacon-outline-ambbpz8h